src/struct_info.rs

```diff
@@ -62,14 +62,16 @@ impl<'a> StructInfo<'a> {
             ..
         } = *self;
         let (impl_generics, ty_generics, where_clause) = self.generics.split_for_impl();
-        let all_fields_param =
-            syn::GenericParam::Type(syn::Ident::new("TypedBuilderFields", proc_macro2::Span::call_site()).into());
+        let empties_tuple = type_tuple(self.included_fields().map(|_| empty_type()));
+        let mut all_fields_param_type: syn::TypeParam =
+            syn::Ident::new("TypedBuilderFields", proc_macro2::Span::call_site()).into();
+        let all_fields_param = syn::GenericParam::Type(all_fields_param_type.clone());
         let b_generics = self.modify_generics(|g| {
-            g.params.insert(0, all_fields_param.clone());
+            all_fields_param_type.default = Some(syn::Type::Tuple(empties_tuple.clone()));
+            g.params.push(syn::GenericParam::Type(all_fields_param_type.clone()));
         });
-        let empties_tuple = type_tuple(self.included_fields().map(|_| empty_type()));
         let generics_with_empty = modify_types_generics_hack(&ty_generics, |args| {
-            args.insert(0, syn::GenericArgument::Type(empties_tuple.clone().into()));
+            args.push(syn::GenericArgument::Type(empties_tuple.clone().into()));
```

Comment:
Note to self: This is a breaking change and should be marked as such in the changelog.